src/lighteval/models/vllm/vllm_model.py

```diff
@@ -300,16 +307,19 @@ def _generate(
         generate: bool = True,
     ) -> list[GenerativeResponse]:
         """Contains the actual logic of the generation."""
+        sampling_params = self.sampling_params.clone() or SamplingParams()
         if generate:
-            sampling_params = SamplingParams(
-                temperature=float(self._config.temperature) if num_samples > 1 else 0.0,
-                n=num_samples,
-                max_tokens=max_new_tokens,
-                stop=stop_tokens,
-                logprobs=1 if returns_logits else 0,
-            )
+            sampling_params.temperature = float(self._config.temperature) if num_samples > 1 else 0.0
```

Comment:
But this is not using the sampling params defined by the user then

Reply:
Ha yes, it overwrites them! True, will need to edit